analyzers/src/SonarAnalyzer.CSharp/Rules/CollectionsShouldImplementGenericInterface.cs

```diff
@@ -34,34 +36,39 @@ public sealed class CollectionsShouldImplementGenericInterface : SonarDiagnostic
         internal const string DiagnosticId = "S3909";
         private const string MessageFormat = "Refactor this collection to implement '{0}'.";
 
-        private static readonly DiagnosticDescriptor rule =
+        private static readonly DiagnosticDescriptor Rule =
             DiagnosticDescriptorBuilder.GetDescriptor(DiagnosticId, MessageFormat, RspecStrings.ResourceManager);
-        public override ImmutableArray<DiagnosticDescriptor> SupportedDiagnostics { get; } = ImmutableArray.Create(rule);
 
-        private static readonly Dictionary<string, KnownType> nongenericToGenericMapping =
-            new Dictionary<string, KnownType>
+        private static readonly Dictionary<string, KnownType> NongenericToGenericMapping =
+            new()
             {
-                [KnownType.System_Collections_ICollection.TypeName] = KnownType.System_Collections_Generic_ICollection_T,
-                [KnownType.System_Collections_IList.TypeName] = KnownType.System_Collections_Generic_IList_T,
-                [KnownType.System_Collections_IEnumerable.TypeName] = KnownType.System_Collections_Generic_IEnumerable_T,
-                [KnownType.System_Collections_CollectionBase.TypeName] = KnownType.System_Collections_ObjectModel_Collection_T,
+                { KnownType.System_Collections_ICollection.TypeName, KnownType.System_Collections_Generic_ICollection_T },
+                { KnownType.System_Collections_IList.TypeName, KnownType.System_Collections_Generic_IList_T },
+                { KnownType.System_Collections_IEnumerable.TypeName, KnownType.System_Collections_Generic_IEnumerable_T },
+                { KnownType.System_Collections_CollectionBase.TypeName, KnownType.System_Collections_ObjectModel_Collection_T },
             };
 
-        private static readonly ImmutableArray<KnownType> genericTypes = nongenericToGenericMapping.Values.ToImmutableArray();
+        private static readonly ImmutableArray<KnownType> GenericTypes = NongenericToGenericMapping.Values.ToImmutableArray();
+
+        public override ImmutableArray<DiagnosticDescriptor> SupportedDiagnostics { get; } = ImmutableArray.Create(Rule);
 
         protected override void Initialize(SonarAnalysisContext context)
         {
             context.RegisterSyntaxNodeActionInNonGenerated(c =>
                 {
-                    var classDeclaration = c.Node as ClassDeclarationSyntax;
+                    if (c.IsRedundantPositionalRecordContext())
```

Comment:
This bail out can be merged with the next bailout. To have only one condition to do it.

Reply:
Why not?

Reply:
We can do it like this, the variables are just renamings, nothing expensive is happening.
```
var typeDeclaration = (BaseTypeDeclarationSyntax)c.Node;
                    var implementedTypes = typeDeclaration.BaseList?.Types;
                    if (implementedTypes == null || c.IsRedundantPositionalRecordContext())
                    {
                        return;
```